pyresample/utils/_cf.py

```diff
@@ -0,0 +1,179 @@
+    try:
+        unit = values.units
+    except AttributeError:
+        unit = 'default'
+
+    # spacing
+    #   TBC: a) should we check the axis has constant spacing or does CF impose this already ?
+    #         b) is a single diff enough, or should we take a mean of several (rounding error) ?
+    spacing = abs(values[1] - values[0])
+
+    # extent (0,5*spacing is because area_def expects corner coordinates of the corner cells,
+    #     while CF stores center coords)
+    extent_low = spacing.min() - 0.5 * spacing
+    extent_hgh = spacing.max() + 0.5 * spacing
+
+    # now we take into account the units
+    scalef = 1.
+    if unit != 'default':
+        if unit == 'km':
+            scalef = 1000.
+        elif unit == 'm':
+            scalef = 1.
+        else:
+            raise ValueError("Sorry: un-supported unit!")
+
+    extent_low *= scalef
+    extent_hgh *= scalef
+
+    return extent_low, extent_hgh, len(values)
+
+
+def load_cf_area(nc_file, variable=None, y=None, x=None, ):
+    """ Load an area def object from a netCDF/CF file. """
+
+    from netCDF4 import Dataset
+    from collections import OrderedDict
+    from pyresample import geometry
+    from pyresample.utils import proj4_str_to_dict
+
+    # basic check on the default values of the parameters.
+    if (x is not None and y is None) or (x is None and y is not None):
+        raise ValueError("You must specify either all or none of x= and y=")
+
+    # the nc_file can be either the path to a netCDF/CF file, or directly an opened netCDF4.Dataset()
+    #   if the path to a file, open the Dataset access to it
+    if not isinstance(nc_file, Dataset):
+        try:
+            nc_file = Dataset(nc_file)
+        except FileNotFoundError:
+            raise ValueError('File not found: {}'.format(nc_file))
+        except OSError:
+            raise ValueError('File is not a netCDF file {}'.format(nc_file))
+
+    # if the variable=None, we search for a good variable
+    if variable is None:
+        variable = 'good variable'
+        raise NotImplementedError("search for a good variable is not implemented yet!")
+    else:
+        # the variable= must exist in the netCDF file
+        if variable not in nc_file.variables.keys():
+            raise ValueError("Variable '{}' does not exist in netCDF file".format(variable))
+
+    # Load a CRS object
+    # =================
+    variable_is_itself_gridmapping = False
+    # test if the variable has a grid_mapping attribute
+    if hasattr(nc_file[variable], 'grid_mapping'):
+        # good. attempt to load the grid_mapping information into a pyproj object
+        crs = _load_crs_from_cf(nc_file, nc_file[variable].grid_mapping)
+    else:
+        # the variable doesn't have a grid_mapping attribute.
+        # ... maybe it is the grid_mapping variable itself?
+        try:
+            variable_is_itself_gridmapping = True
+            crs = _load_crs_from_cf(nc_file, variable)
+        except Exception as ex:
```

Comment:
Can you be more specific on which error(s) to expect?

Reply:
Yes. `pyproj.exceptions.CRSError`